articles/140_topic_and_service_name_mapping.md

```diff
@@ -391,6 +338,31 @@ This compromise was made so that when more work is done on substitutions, it wou
 
 There was some discussion of alternatives and concerns with respect to the ROS -> DDS translation.
 
+#### Alternative using DDS Partitions
+
+Previously the usage of forward slashes (`/`) was disallowed in DDS topic name and hence a strategy was proposed which used DDS partitions to address the forward slashes (`/`) which are present in ROS names. The main idea was to separate the ROS name into the "namespace" and the "base name", and then place the namespace, stripped of leading and trailing forward slashes (`/`), into a single DDS partition entry and the remaining base name into the DDS topic name.
+This addressed the issue because the ROS name's base name will not contain any forward slashes (`/`) by definition and so there are no longer any disallowed characters in the DDS topic name.
+The DDS partition would contain the ROS name's namespace, including any forward slashes (`/`) that made up the namespace and were not at the beginning or the end of the namespace.
+That is acceptable because DDS partitions are allowed to contain forward slashes (`/`) unlike the DDS topics previously but now DDS topic names allow forward slashes (`/`).
+
+DDS partitions are implemented as an array of strings within the `DDS::Publisher` and `DDS::Subscriber` QoS settings and have no hierarchy or order, Each entry in the partition array is directly combined with the DDS topic and they are not sequentially combined.
+If a publisher has two partition entries, e.g. `foo` and `bar` with a base name of `baz`, this would be equivalent to having two different publishers on these topics: `/foo/baz` and `/bar/baz`.
+Therefore this proposal used only one of the strings in the partitions array to hold the entire ROS name's namespace.
+
+You can read more about partitions in RTI's documentation:
+
+- [PARTITION_QosPolicy](https://community.rti.com/static/documentation/connext-dds/5.2.3/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/Content/UsersManual/PARTITION_QosPolicy.htm)
+
+Trade-offs (in comparison to using the whole ROS name along with the namespaces):
+
+- Splitting the ROS name into "namespace" and "base name", and placing the complete namespace into a single partition seemed un-natural.
```

Comment:
why unnatural?
Another reason for this is that newer standards such as XRCE DDS might not have partitions at all.

Reply:
maybe there is a better word that un-natural. Placing the whole namespace with the prefix in a 'single' partition sounds like a workaround and therefore I added the term un-natural